code/_onclick/click.dm

```diff
@@ -129,7 +129,7 @@
 
 	// Allows you to click on a box's contents, if that box is on the ground, but no deeper than that
 	if(isturf(A) || isturf(A.loc) || (A.loc && isturf(A.loc.loc)))
-		if(A.Adjacent(src)) // see adjacent.dm
+		if(A.Adjacent(src) || W.reach >= get_dist(src, A)) // see adjacent.dm

```

Comment:
Hmm, I do see however that adjacency would always win the 0 case, if that's what you were trying to say. (The if will always succeed on a same-tile 0 range weapon)
You could snowflake a reach of 1 to be the only case that runs ```Adjacent()```, if you -really- want 0 reach weapons.

```
if(W.reach == 1 && A.Adjacent(src))
    ...
else if(W.reach >= get_dist(src, A))
    ...
```